### PR TITLE
Add optional configuration parameter to select JSON schema version

### PR DESCRIPTION
**Pull request by hpmellema**
*Issue #, if available:* [issue #1252](https://github.com/awslabs/smithy/issues/1252)

*Description of changes*:
Adds an optional configuration parameter `jsonSchemaVersion` to select the version of JSON schema to use. The JSON schema version will still default to `draft07`, but can be set to `draft2020-12`. This new configuration parameter will also provide a mechanism to support future JSON schema changes.

Reviewing the JSON schema specifications, it appears that the only change needed was to update "definitions" -> "$defs" as Smithy jsonSchema does not support meta-schema generation, the use of plain-name fragments, tuples, or dynamic anchors.

*Cloudformation*
Cloudformation only supports jsonSchema `draft07` at this time (see [here](https://github.com/aws-cloudformation/cloudformation-template-schema/blob/25fdf51f246fff8129539fc8a19876bf9dc197cd/src/main/java/aws/cfn/codegen/json/SchemaDraft.java))
In order to prevent compatibility issues, a check was added to the cloudformation plugin to ensure only JSON Schema `draft07` is used in the configuration. I expect we will update this as Cloudformation and API Gateway add support for later versions.

*OpenAPI*
The OpenAPI plugin was updated to set the JSON Schema version parameter based on the latest version supported by the OpenAPI spec version. 
I.e. OpenAPI3.1 will use `draft2020-12` while OpenAPI3.0 will use `draft07`

*Testing* 
- manually tested that both OpenAPI and Cloudformation plugins correctly used the new configuration parameter and rejected unsupported versions/parameters
- Added a new integration test for `draft2020-12` in the JSON schema module
- Added test that OpenAPI config correctly associates with the right JSON schema version

*JSON Schema Change list*
The following two web pages detail the 
Schema Changes: 
- [draft07 -> draft2019-09](https://json-schema.org/draft/2019-09/release-notes.html)
- [draft2019-09 -> draft2020-12](https://json-schema.org/draft/2020-12/release-notes.html)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaConfig.java [hunk 1]
```diff
@@ -403,4 +404,24 @@ public boolean getUseIntegerType() {
     public void setUseIntegerType(boolean useIntegerType) {
         this.useIntegerType = useIntegerType;
     }
+
+    /**
+     * Set the JSON schema version to use when converting Smithy shapes into Json Schema.
```
Comment (srchase) [inline smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaConfig.java:409]: nit: This looks looks like the text for the setter.

File: smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaVersion.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (srchase) [inline smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaVersion.java:2]: nit: Wrong year. [This](https://github.com/awslabs/smithy/pull/1614) updated checkstyle so we can use the short-form license header without the year now:

```
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
```

Comment (hpmellema) [reply smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaVersion.java]: updated in cr 3f12c5b.

File: smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaConfig.java (hunk 1 above)
Comment (hpmellema) [reply smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/JsonSchemaConfig.java]: updated in cr 3f12c5b.

**Review by srchase — APPROVED** [review]

**Pull request merged**